src/lib/parser.rs

```diff
@@ -238,25 +238,32 @@ impl Parser {
       return Ok(());
     }
     self.index -= 1;
-    match self.try_match(&[(Keywords::Fn, " \t\n"), (Keywords::Const, " \t\n")]) {
-      Some(Keywords::Const) => {
-        let parsed_variable = parse_var(self, Some(VarType::Const))?;
-        self.global_vars.push(parsed_variable);
-        Ok(())
-      }
-      Some(Keywords::Fn) => {
-        let parsed_function = ParseFunction::start(self)?;
-        self.functions.push(parsed_function);
-        Ok(())
-      }
-      _ => {
-        if let Some(c) = self.next_char() {
-          self.unexpected_char(c)
-        } else {
-          self.unexpected_eof()
+    while self.index < self.contents.len() {
+      match self.try_match(&[(Keywords::Fn, " \t\n"), (Keywords::Const, " \t\n")]) {
+        Some(Keywords::Const) => {
+          let parsed_variable = parse_var(self, Some(VarType::Const))?;
+          self.global_vars.push(parsed_variable);
+          continue;
```

Comment:
This continue is not really needed right because there won't be any code executed after this expect going to the next loop.  
```suggestion
```